Hw5/Task1/SparseMatrix.cs

```diff
@@ -0,0 +1,120 @@
+using System.Collections;
+using System.Text;
+
+public class SparseMatrix : IEnumerable<long>
+{
```

Comment:
1. Maybe not effective from performance point of view. You save memory, but will use ToFlattenIndex() each time you access  data. Better to use Dictionary<int, int, long> nonZero, more simple logics and no conversions from int to (int, int) and vice versa.
2. Consider situation -> Height = Int.MaxValue, Wedth int.MaxValue; Your int  FlattenIndex will be overflowed in checked context, or, worse, unexpected behavior in unchecked context  (which is used as default). Please fix, at least use  Dictionary<long, long>